### ISSUE TITLE
How can i use ESP8266 as a WI-FI adaptor?

### ISSUE THREAD

**Issue by the-hk** [tttapa/ESP8266#81]
i mean is there any code or way to connect my computer which has not wifi module to connect my modem and i have esp8266 and i would like to use this module as a wifi module is it possible?

**Comment by tttapa** [comment]
I'm sure it's possible, but it's probably not worth the time investment. Especially since USB WiFi dongles are so cheap.

**Comment by aramperez** [comment]
Someone correct me, but I don't believe it is possible. When the ESP8266 is in AP mode, it can NOT connect to the Internet.

**Comment by tttapa** [comment]
> Someone correct me, but I don't believe it is possible. When the ESP8266 is in AP mode, it can NOT connect to the Internet.

Why would the ESP8266 need to be in AP mode?  
Also, you can use the WiFi in AP and station mode at the same time: https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/generic-class.html?highlight=WIFI_AP_STA#mode

**Comment by aramperez** [comment]
Yes, but I would not recommend it, search for "esp8266 sta ap channel issues".

And as you originally stated, "it's probably not worth the time investment".

**Comment by juancb** [comment]
What about using it as a wifi interface for another microcontroller board like an Arduino UNO R3?
